src/urllib3/util/ssl_.py

```diff
@@ -35,7 +33,7 @@ def _is_ge_openssl_v1_1_1(
     )
 
 
-try:  # Do we have ssl at all?
+if TYPE_CHECKING:  # Assume ssl module is there.
```

Comment:
We don't want these imports to be predicated on type checking being active. We should move the `if TYPE_CHECKING` to only cover the `typing_extensions` import.

Reply:
Refactored it and it uses some `# type: ignore`s for now, not sure if this is ok in this case